tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,143 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
+	}
+	misc.LogTestEnd(c.TestName())
+}
+
+// test to verify volume is removed after user created vmgroup is removed without "--remove-volume" option
+// 1. create a user created vmgroup "vmgroup_test1" and add VMs to this vmgroup
+// 2. create two volumes for this vmgroup, one is on default datastore, and the other is on non-default datastore
+// 3. verify mount/unmount for this volume
+// 4. after unmount, verify the volume is detached for both volumes
+// 5. remove VMs from this vmgroup, and remove the vmgroup without "--remove-volume" option
+// 6. verify that volumes exist but do not belong to any vmgroup
+func (vg *VmGroupTest) TestVmgroupRemove(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume in vgTestVMgroup1
+	vg.createVolumes(c, vg.volName1)
+
+	// Add create privilege for the second datastore
+	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create another volume on second datastore
+	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+	// Verify volume can be mounted and unmounted for the first volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+	// Verify volume can be mounted and unmounted for the second volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status = verification.VerifyDetachedStatusOnNonDefaultDS(vg.volName2, vg.config.Datastores[1],
+		vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName2+"@"+vg.config.Datastores[1]))
+
+	// remove VM1 and VM2 from vgTestVMgroup1 and then remove the vmgroup
+	vmList := []string{vg.config.DockerHostNames[0], vg.config.DockerHostNames[1]}
+	for _, vm := range vmList {
+		out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vm)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+
+	log.Printf("Removing test vmgroup %s", vgTestVMgroup1)
+	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, false)
+	c.Assert(err, IsNil, Commentf(out))
+
+	volumeList := []string{vg.volName1, vg.volName2}
+	for _, volume := range volumeList {
+		// volume does not belong to any vmgroup, so "out" is expected to be "N/A"
+		out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, volume)
+		log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, volume)
+		c.Assert(out, Equals, "N/A", Commentf("volume %s should not belong to any vmgroup", volume))
+	}
+
```

Comment:
Those volumes are no orphan volumes, and user have no way to access/cleanup those orphan volumes. So we cannot cleanup them through adminCLI or dockerCLI.
As we dicussed in https://github.com/vmware/docker-volume-vsphere/issues/1469, we should NOT allow user to remove a vmgroup if volumes still in the vmgroup. So testing the scenario that "remove vmgroup without removing volumes" is not needed if we decided to not support remove vmgroup without removing volumes. 

I plan to remove this test from the PR.